### PR TITLE
[SDTEST-878] Optimise LocalRepository.relative_to_root helper to make test impact analysis faster

### PR DESCRIPTION
**Pull request by anmarchenko**
**What does this PR do?**
Applies some heuristics to avoid calling `Pathname#relative_path_from` function when determining what would be path relative to the git root for the given filepath:
- If given filepath is absolute, we use the fact that absolute paths are pre-filtered already and they always contain root as prefix, so we only need to slice the string to get the suffix (`path[prefix_length..]`)
- If given filepath is relative, we compute `Pathname#relative_path_from` once and then we store the difference between the relative filepath and the computed relative path from root in the `@prefix_to_root` cached variable. We can do it because all relative paths are computed relative to the `Dir.pwd` which doesn't change in the lifetime of the process. After we have `@prefix_to_root` cache, we prepend it to every filepath we get (or just return the filepath if the tests are running from the git root).

**Motivation**
Profiling rubocop test suite with Datadog continuous profiler showed that big part of CPU time is spent in `Datadog::CI::Git::LocalRepository.relative_to_root` function:
profile-before

**How to test the change?**
Unit tests are provided.

[Benchmark results](https://github.com/DataDog/test-environment/pull/329):
- Middleman overhead 34,1% (down from 40%)
- Rubocop overhead 64% (down from 128%) - **2x improvement**

### CODE REVIEW

File: lib/datadog/ci/transport/http.rb [hunk 1]
```diff
@@ -149,6 +149,10 @@ def code
             nil
           end
 
+          def response_size
+            0
```
Comment (anmarchenko) [inline lib/datadog/ci/transport/http.rb:153]: unrelated change, I noticed that library started to fail if certain backend API is not available because for ErrorResponse there is no http_response

**Review by juan-fernandez — APPROVED** [review]
